packages/react/src/bags/hooks/useBag.ts

```diff
@@ -300,6 +300,10 @@ const useBag = (options: UseBagOptions = {}) => {
     (
       productId: Product['result']['id'],
       { quantity, sizeId }: { quantity: number; sizeId: number },
+      productAggregatorId?: Exclude<
+        BagItemDenormalized['productAggregator'],
+        null
+      >['id'],
```

Comment:
Adding the `productAggregatorId` as a parameter will force all the functions that use the `metadata` and/or `config` parameters to change to add `undefined` on their calls.

If the second parameter of `addItem` accepted the new optional parameter `productAggregatorId`, this new change would not force anyone to change the addItem callback:

```tsx
const addItem = useCallback(
    (
      productId: Product['result']['id'],
      {
        quantity,
        sizeId,
        productAggregatorId,
      }: {
        quantity: number;
        sizeId: number;
        productAggregatorId:
          | Exclude<BagItemDenormalized['productAggregator'], null>['id']
          | undefined;
      },
      metadata?: BagItemActionMetadata,
      config: Config | undefined = fetchConfig,
    )
// ...
```

Reply:
I agree with @dnamorim and I would also add that the types of productId, quantity, sizeId, productAggregatorId and metadata can be derived from the `PostBagItemData` type as they are ultimately based on.

Reply:
Great catch! Thanks, updated 👍